TheMovies/TheMovies/Services/APICall.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Newtonsoft.Json;
+using System;
+//using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+
+namespace TheMovies.Services
+{
+    class APICall
+    {
+        public static readonly string TMDbToken = "1f54bd990f1cdfb230adb312546d765d";
+        public static readonly string ApiUrl = "https://api.themoviedb.org/3/{0}/";
+        //public static readonly string ApiUrl = "https://api.themoviedb.org/3/movie/";
+        public static readonly string ApiUrlForImage = "http://image.tmdb.org/t/p/";
+
+        public APICall()
+        {
+
+        }
+
+        // ToDo: We can change the 2nd parameter from string to dictionary
+        public async Task<T> GetResponse<T>(string target, string method, string param) where T : class
```

Comment:
All the comments in this class make it difficult to read the code. Some refactoring could make most of the comments obsolete and improve readability.